api/buckets/service.go

```diff
@@ -1913,15 +2287,38 @@ func (s *Service) pinBlocks(ctx context.Context, nodes []ipld.Node) error {
 // sumBytesPinned adds the provided delta to the buckets total size from
 // the account/user.
 func (s *Service) sumBytesPinned(ctx context.Context, delta int64) error {
-	a := accountFromContext(ctx)
+	var a *mdb.Account
+	var u *mdb.User
+	if owner, ok := ctx.Value(ctxKey("owner")).(string); ok && owner != "" {
```

Comment:
This is a bit of a mess, but the idea is to target the storage for the bucket owner, not necessarily the account/user that is making the change.

Reply:
Q: So the owner of the bucket is always loaded in context in the interceptor?

Should the below `getOwnerFromContext` be renamed? Doesn't look like returning doing the same logic or using "owner" in another sense?

Reply:
Any API method that can effect bucket size attaches the "owner" directly (not in the interceptor), like
```
ctx = context.WithValue(ctx, ctxKey("owner"), buck.Owner)
```
I went that route instead of passing around the bucket owner string to all these pinning, unpinning, node adding, etc. methods, which looked pretty nasty.

Reply:
OK! Do you have any comments about `getOwnerFromContext` possible renaming?
Shouldn't be `getUAccountOrUserFomContext` instead of using the "owner" concept? 
Maybe I'm confused!

Reply:
Good call, I'll rename `getOwnerFromContext` to be more clear, maybe `getAccountOrUserFromContext`. It will be nice to remove the `User` abstraction and make it a special type of `Account`. I think that alone would help clean this up.

Reply:
Oop, we posted at the same time. SGTM!